src/lib/annotations/doc/DocAnnotator.js

```diff
@@ -834,59 +806,39 @@ class DocAnnotator extends Annotator {
         let consumed = false;
         let activeThread = null;
 
-        // Destroy any pending highlights on click outside the highlight
-        const pendingThreads = this.getThreadsWithStates(PENDING_STATES);
-        pendingThreads.forEach((thread) => {
-            if (thread.type === TYPES.point) {
-                thread.destroy();
-            } else {
-                thread.cancelFirstComment();
-            }
-        });
-
-        // Only filter through highlight threads on the current page
         const page = annotatorUtil.getPageInfo(event.target).page;
-        const pageThreads = this.getHighlightThreadsOnPage(page);
+        const pageThreads = this.threads.get(page);
+        if (!pageThreads) {
+            return;
+        }
+
         pageThreads.forEach((thread) => {
-            // We use this to prevent a mousedown from activating two different
-            // highlights at the same time - this tracks whether a delegated
-            // mousedown activated some highlight, and then informs the other
-            // keydown handlers to not activate
-            const threadActive = thread.onClick(event, consumed);
-            if (threadActive) {
-                activeThread = thread;
-            }
+            if (PENDING_STATES.indexOf(thread.state) > 0) {
+                // Destroy any pending highlights on click outside the highlight
+                if (thread.type === TYPES.point) {
+                    thread.destroy();
+                } else {
+                    thread.cancelFirstComment();
+                }
+            } else if (annotatorUtil.isHighlightAnnotation(thread.type)) {
+                // We use this to prevent a mousedown from activating two different
+                // highlights at the same time - this tracks whether a delegated
+                // mousedown activated some highlight, and then informs the other
+                // keydown handlers to not activate
+                const threadActive = thread.onClick(event, consumed);
+                if (threadActive) {
+                    activeThread = thread;
+                }
 
-            consumed = consumed || threadActive;
+                consumed = consumed || threadActive;
+            }
         });
 
         // Show active thread last
         if (activeThread) {
             activeThread.show();
         }
     }
-    /**
-     * Returns all threads with a state in the specified states.
-     *
-     * @private
-     * @param {...string} states - States of highlight threads to find
-     * @return {AnnotationThread[]} threads with the specified states
-     * */
-    getThreadsWithStates(...states) {
```

Comment:
This was also being used only once where we would end up iterating through ALL the file version's annotations once to generate the array in this method and then iterate through that array again wherever getThreadsWithStates() was being used.